docs/architecture-decisions/2023-06-06_frontend-templates.md

```diff
@@ -0,0 +1,133 @@
+
+This orthogonal approach proposes to improve capabilities of the algokit-cli codebase by making it adaptable to various frontend stacks to allow anyone to easily convert their existing frontend projects into web3 dApps.
+
+The implementation specific details will consist of deriving a set of bare minimum requirements for such feature to scan and understand the structure of frontend where algokit is being embedded and performing necessary modifications to project's files. A detailed discussion can be held in a scope separate from this Architecture Decision Record.
+
+> This approach can be explored and maintained without overlapping with main proposal on frontend templates.
+
+### Addendum 2. Website for choosing preferred frontend and backend repositories.
+
+> TLDR: The approach proposes a website to enhance the discoverability of official and community-based algokit templates, thereby incentivizing template builders to create and maintain their own templates.
+
+This orthogonal approach proposes to improve discoverability of official and community based algokit templates by providing a simple static website. The website can consist of minimalistic UI components for picking preffered backend, frontend and then a `Generate` button that will output copy-pasteable algokit CLI commands to spin up a project with the selected templates.
+
+As a specific example, the website can be hosted on [AwesomeAlgo](https://awesomealgo.com) website, thus ensuring that this is an open-source community maintaned entrypoint for discovering and using algokit templates. Removing the need and maintenance overhead on our teams to maintain it as official resource.
+
+Lastly, community template builders will get a platform to increase discoverability of their templates and further incentivise them to build and maintain them. While developers using the templates can support creators of templates by donating to their projects (a simple tipping mechanism for Algo and ASAs can be embedded into the website) or by contributing to the templates themselves.
+
+> This approach can be explored and maintained without overlapping with main proposal on frontend templates.
+
+---
+
+## Final decision
+
+To be decided
+
+> The final decision is being finalized but the team is currently leaning in favor towards the proposal outlined in the Architecture Decision Record. Hence, this revision excluded the details of other options considered and adds more decision specific details to the approach with having separate self-sufficient frontend templates.
+
+## Open questions
+
+- Python typed clients are not going to be too relevant for scenarios where they need to be integrated with frontends given that in majority of cases a developer would prefer a TS typed client. Hence, how do we ensure that user gets a clear information and indication on when to use Python typed clients vs TS typed clients?
```

Comment:
This is a fair comment, but to my mind it's a clear dilineation in that it's dependant on what language you are using.

While there are Python website libraries, we aren't using them we will use JavaScript (TypeScript).